### PR TITLE
Forward-substitute long calls on 32 bit targets

### PR DESCRIPTION
**Pull request by SingleAccretion**
The multi-reg restrictions do not apply to them.

[Diffs](https://dev.azure.com/dnceng/public/_build/results?buildId=1965088&view=results) - a few improvements as expected.

### CODE REVIEW

File: src/coreclr/jit/forwardsub.cpp [hunk 1]
```diff
@@ -673,32 +673,24 @@ bool Compiler::fgForwardSubStatement(Statement* stmt)
...
         }
 
-#if defined(TARGET_X86) || defined(TARGET_ARM)
-        if (fwdSubNode->TypeGet() == TYP_LONG)
-        {
-            JITDUMP(" TYP_LONG fwd sub node, target is x86/arm\n");
-            return false;
-        }
-#endif // defined(TARGET_X86) || defined(TARGET_ARM)
-
```
Comment (JulieLeeMSFT) [inline src/coreclr/jit/forwardsub.cpp:701]: Just a question. Why do we delete this code that skips TYP_LONG forward substitute node on x86 and arm target?

Comment (SingleAccretion) [reply src/coreclr/jit/forwardsub.cpp:701]: It should not be necessary.

The restriction on substituting multi-reg nodes stems from the fact the backend (LSRA and codegen) can only handle a few select forms of uses for them:
1) `STORE_LCL_VAR(multi-reg node)`
2) `RETURN(matching multi-reg call | LCL_VAR)`

For structs, these restrictions are enforced by the frontend (mainly importer). For `LONG`s - by decomposition, which means we don't need to worry about them here.

**Comment by SingleAccretion** [comment]
@dotnet/jit-contrib

**Review by AndyAyersMS — APPROVED** [review]

**Comment by AndyAyersMS** [comment]
Thanks again for all your hard work.

**Pull request merged**